alsrb0504/20230523/카잉달력/카잉달력.js

```diff
@@ -0,0 +1,56 @@
+const readFileSyncAddress =
+  process.platform === "linux" ? "/dev/stdin" : "text.txt";
+
+const input = require("fs")
+  .readFileSync(readFileSyncAddress)
+  .toString()
+  .trimEnd()
+  .split("\n");
+
+const answer = [];
+const N = Number(input[0]);
+
+for (let i = 1; i <= N; i++) {
+  answer.push(solution(i));
+}
+
+console.log(answer.join("\n"));
+
+function solution(line) {
+  const [M, N, X, Y] = input[line].split(" ").map(Number);
+
+  // 예외 : X와 Y가 같으면 바로 리턴 가능
+  if (X === Y) return X;
+
+  // 풀이 방법
+  // X를 고정해두고 Y의 값을 M씩 증가시키며 가능한 최소값 찾기.
+  // Y를 고정해두고 X의 값을 N씩 증가시키며 앞서 구한 최소값과
+  // 이번에 구한 최소값 중 더 작은 값 리턴.
+
+  //  min: X 고정시의 최소값을 저장할 변수
+  let min = Infinity;
+  let cnt = X;
+
+  while (cnt <= N * M) {
+    cnt += M;
+    const rest = cnt % N === 0 ? N : cnt % N;
+
+    if (rest === Y) {
+      min = cnt;
+      break;
+    }
+  }
+
+  cnt = Y;
+
+  while (cnt <= N * M) {
+    cnt += N;
+    const rest = cnt % M === 0 ? M : cnt % M;
+
+    if (rest === X) {
+      return Math.min(cnt, min);
+    }
+  }
```

Comment:
```suggestion
    if (rest === Y) {
      return cnt;
    }
  }
```

제 방법대로 라면 이렇게 하면 통과가 되어야하는데.. 이 코드에선 또 안되네요...? 두 케이스 중 최소값을 찾아주신 이유가 있을까요?? 저는 X,Y를 충족하는 cnt는 하나밖에 없다고 생각해서 cnt를 찾자마자 리턴을 해주었어요!

Reply:
처음에 `cnt = x`을 할당한 후, cnt의 값을 `+= M` 씩 증가시키며 찾아봤는데 지은님의 리뷰에 남겼었던 이 경우가 해결이 안 되어서 두 케이스 중 최소값을 찾았습니다.
```
// 저는 이 경우 때문에 cnt = y로 설정한 후, cnt += N 을 실행했었습니다..!
input = [60 15 20 5]
answer = 20
```

방금까지 지은님의 코드를 일부 가져와서 이리저리 만져봤는데.. 여전히 틀리네요.. ㅠㅠ 아마 처음부터 코드를 다시 작성하면서 틀린 곳을 찾아봐야 할 것 같습니다 😥